### PR TITLE
Port do_after improvements

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Ports do_after improvements from TG.
Ports actionspeed modifiers from TG (will be implemented in #1226).
Ports TG sprites.
Ports interactive do_afters.
Also I went through all do_afters and fixed em up.

Credit to TG and Mojave Sun (technically).

## Why It's Good For The Game

Alright alright I get it, I can hear it already, but trust me there's a few good reasons to have interactive do_afters

1. Requires people to actually focus on things that would normally require focus. IRL it's pretty hard to operate a control panel while also staring at a massive statue, so why is it so easy in-game?
2. Gives us more variables to tweak with do_afters. It's not just "you go faster or slower", you can manipulate reward values, QTE difficulty, speed, hell you could even have wacky things (being high, going insane, memetic agents) have wacky effects (moving targets, fake targets, alternate coloring, etc).
3. Lets us add in more interesting gameplay for do_afters. Loads of servers have do_afters where you want to do something that someone else wants you to NOT do (forcefeeding, breaking handcuffs, etc), and usually the only real "game" there is to swap places with them and they magically stop (both unrealistic and quite boring). Instead, we can try and have mechanics where you actually have to STOP the person trying to inject you instead of just abusing the system in the dumbest way possible.

## Changelog

:cl:
add: Interactive do_afters! During (most) timed actions, stars will pop up that can be clicked to get extra progress.
add: Actionspeed modifiers ported from TG.
Balance: do_afters using the interactive system have had default times increased to account for extra progress.
imageadd: do_after sprites ported from TG.
imageadd: Interactive do_after sprites by MrDoomBringer.
refactor: Massive improvements to do_after system (from TG).
/:cl:

### CODE REVIEW

**Comment by TheDarkElites** [comment]
ugh I dont wanna review this but I guess I have to

**Review by TheDarkElites — DISMISSED** [review]
didn't check all do_after calls, but the do_after and associated code itself seem fine to me. Will have Tsurpta look over it before the proper merge.

File: code/datums/progressbar.dm [hunk 1]
```diff
@@ -2,119 +2,228 @@
...
+	var/bonus_progress = 0
+
+/datum/progressbar/New(mob/_user, goal_number, atom/_target, _show_target = FALSE, bonus_percentage, focus_sound, focus_frequency)
+	. = ..()
+	if (!istype(_target))
+		CRASH("Invalid target [_target] passed in")
+	if(QDELETED(_user) || !istype(_user))
+		CRASH("a /datum/progressbar was created with [isnull(_user) ? "null" : "invalid"] user")
+	if(!isnum(goal_number))
+		CRASH("a /datum/progressbar was created with [isnull(goal_number) ? "null" : "invalid"] goal_number")
```
Comment (Tsurupeta) [inline code/datums/progressbar.dm]: replace crash() with stack_trace, we have something like auxtools_stack_trace or crash_with (literally same thing) and add back qdel(src)

Comment (aidan-bowman) [reply code/datums/progressbar.dm]: done

**Pull request closed without merging**